### PR TITLE
Fix broken unit tests

### PR DESCRIPTION
**Pull request by jdorweiler**
**Reviewer:** @dharb 

## Description:

Fixed all the broken unit test and added a basic tracker blocking test. 
Fixed linter errors

## Steps to test this PR:
1. All tests should be passing
2. Everything else should work as expected. Try some sites and make sure trackers are blocked and popup looks okay. 

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/utils.es6.js [hunk 1]
```diff
@@ -3,38 +3,6 @@ const tdsStorage = require('./storage/tds.es6')
 const constants = require('../../data/constants')
 const parseUserAgentString = require('../shared-utils/parse-user-agent-string.es6')
 const browserInfo = parseUserAgentString()
-const load = require('./load.es6')
-
-/* Check to see if a company is related to a domain.
- * @param {string} entity - company name
- * @param {string} domain - domain or url
- */
-function isRelatedEntity (entityName, domain) {
```
Comment (jdorweiler) [inline shared/js/background/utils.es6.js]: Not used anymore

File: shared/js/ui/models/site.es6.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 const Parent = window.DDG.base.Model
 const constants = require('../../../data/constants')
-const trackerPrevalence = require('../../../data/tracker_lists/prevalence')
```
Comment (jdorweiler) [inline shared/js/ui/models/site.es6.js]: This is in the tds data now

File: unit-test/background/classes/site.es6.js [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 
 const EXT_ID = `ogigmfedpbpnnbcpgjloacccaibkaoip`
-const tempWhitelist = ['suntrust.com', 'onlinebanking.nationwide.co.uk', 'accounts.google.com']
 
 describe('Site', () => {
-    beforeEach(() => {
+    beforeAll(() => {
+        load.loadStub({brokenSites: fakeBrokenSites})
         spyOn(browserWrapper, 'getExtensionId').and.returnValue(EXT_ID)
-        spyOn(abpLists, 'getTemporaryWhitelist').and.returnValue(tempWhitelist)
```
Comment (jdorweiler) [inline unit-test/background/classes/site.es6.js]: temp whitelist is part of tdsStorage now

File: integration-test/background/grades.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ const tests = [
     { url: 'duckduckgo.com', siteGrade: 'A', enhancedGrade: 'A' },
     { url: 'theguardian.com', siteGrade: 'D-', enhancedGrade: 'B+' },
     { url: 'google.com', siteGrade: 'D', enhancedGrade: 'D' },
-    { url: 'amazon.com', siteGrade: ['D-', 'C+'], enhancedGrade: 'C+' },
```
Comment (kdzwinel) [inline integration-test/background/grades.js]: We allow enhanced grade (C) to be < than the default grade (C+)? 🤔

File: shared/js/ui/models/site.es6.js (hunk 2 above)
Comment (kdzwinel) [reply shared/js/ui/models/site.es6.js]: Interesting that it was included w/o being used 🤷‍♂

File: unit-test/data/tdsTestDomains.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[
+    {"tracker": "https://google-analytics.com/ga.js", "site": "reddit.com", "req": {"type": "script"}, "result": {"action": "redirect", "firstParty": false, "reason": "matched rule - surrogate"}},
+    {"tracker": "https://google-analytics.com/ga.js", "site": "google.com", "req": {"type": "script"}, "result": {"action": "ignore", "firstParty": true, "reason": "first party"}}
+]
```
Comment (kdzwinel) [inline unit-test/data/tdsTestDomains.json]: I'd be cool to add couple more (e.g. facebook) that we know should be blocked (we only cover redirect and ignore above).

File: integration-test/background/grades.js (hunk 4 above)
Comment (jdorweiler) [reply integration-test/background/grades.js]: Good catch. I'll try removing C+ and see how it goes.

**Pull request merged**